packages/@react-aria/calendar/src/useCalendarBase.ts

```diff
@@ -39,36 +39,42 @@ export function useCalendarBase(props: CalendarPropsBase & DOMProps, state: Cale
   // Kinda hacky, but it's the easiest way to get the id from the cell.
   calendarIds.set(state, calendarId);
 
+  let focusFocusedDateCell = useCallback(() => {
+    const focusedCell = calendarBody.current.querySelector(`#${getCellId(state.focusedDate, calendarId)}`);
+    if (focusedCell && focusedCell !== document.activeElement) {
+      focusedCell.focus();
+    }
+  }, [state.focusedDate, calendarId]);
+
   useEffect(() => {
     // focus the calendar body when mounting
     if (autoFocus) {
-      calendarBody.current.focus();
+      focusFocusedDateCell();
     }
   }, [autoFocus]);
 
   // Announce when the current month changes
   useUpdateEffect(() => {
-    announce(monthFormatter.format(state.currentMonth));
+    if (state.isFocused) {
```

Comment:
comment explaining why we focusFocusedDateCell if the state is already focused?

Reply:
`state.isFocused` is for the case where month has changed and focus is within the calendar grid. Like either `PageUp` or `PageDown` or navigating from one month to the next using the arrow keys. The other case is when month changes after the user presses the Next or Previous Month button.

Reply:
great, i would like that comment in the code :)